File: Practica 3/Cola.java
import java.util.Scanner;

public class Cola {
    private long[] arreglo;
    private int inicio, fin, n, cantidadElementos;

    public Cola(int n) {
        this.n = n;
        arreglo = new long[n];
        inicio = 0;
        fin = -1;
        cantidadElementos = 0;
    }

    public void insert(long e) {
        if (isFull()) {
            System.out.println(" Cola llena.");
            return;
        }
        fin = (fin + 1) % n;
        arreglo[fin] = e;
        cantidadElementos++;
        System.out.println(" Insertado: " + e);
    }

    public long remove() {
        if (isEmpty()) {
            System.out.println(" Cola vacía.");
            return -1;
        }
        long elemento = arreglo[inicio];
        inicio = (inicio + 1) % n;
        cantidadElementos--;
        System.out.println(" Eliminado: " + elemento);
        return elemento;
    }

    public long peek() {
        return isEmpty() ? -1 : arreglo[inicio];
    }

    public boolean isEmpty() {
        return cantidadElementos == 0;
    }

    public boolean isFull() {
        return cantidadElementos == n;
    }

    public int size() {
        return cantidadElementos;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("tamaño de cola: ");
        int tamano = scanner.nextInt();
        Cola cola = new Cola(tamano);

        cola.insert(10);
        cola.insert(20);
        cola.insert(30);
        cola.insert(40);
        cola.insert(50);

        System.out.println(" Primer elemento (peek): " + cola.peek());
        cola.remove();
        cola.remove();
        System.out.println(" Tamaño actual de la cola: " + cola.size());
        cola.insert(60);
        cola.insert(70);

        scanner.close();
    }
}

File: Practica 3/Cola.py
class Cola:
    def __init__(self, n):
        self.n = n
        self.arreglo = [0] * n
        self.inicio = 0
        self.fin = -1
        self.cantidad_elementos = 0

    def insert(self, e):
        if self.isFull():
            print("Cola llena")
            return
        self.fin = (self.fin + 1) % self.n
        self.arreglo[self.fin] = e
        self.cantidad_elementos += 1

    def remove(self):
        if self.isEmpty():
            print("Cola vacía")
            return -1
        elemento = self.arreglo[self.inicio]
        self.inicio = (self.inicio + 1) % self.n
        self.cantidad_elementos -= 1
        return elemento

    def peek(self):
        if self.isEmpty():
            print("Cola vacía.")
            return -1
        return self.arreglo[self.inicio]

    def isEmpty(self):
        return self.cantidad_elementos == 0

    def isFull(self):
        return self.cantidad_elementos == self.n

    def size(self):
        return self.cantidad_elementos

if __name__ == "__main__":
    cola = Cola(5)
    cola.insert(10)
    cola.insert(20)
    cola.insert(30)

    print("Primer elemento (peek):", cola.peek())
    print("Eliminado:", cola.remove())
    print("Tamaño actual:", cola.size())
